packages/block-editor/src/components/block-list/style.scss

```diff
@@ -104,8 +100,29 @@
 			border-color: var(--wp-admin-theme-color);
 		}
 	}
-}
 
+	// Ensure an accurate partial text selection.
+	// To do this, we disable text selection on the main container, then re-enable it only on the
+	// elements that actually get selected.
+	// To keep in mind: user-select is currently inherited to all nodes inside.
+	user-select: none;
+
+	// Re-enable it on components inside.
+	[class^="components-"] {
+		user-select: text;
+	}
+
+	// Re-enable it on editable items and a few others.
+	.block-editor-block-list__block,
+	[contenteditable] {
+		user-select: text;
+
+		// Hide the select style pseudo element as it interferes with the style.
+		&.is-partially-selected::after {
+			display: none;
```

Comment:
This creates a regression when we have multiselected blocks and hover the block toolbar. This is in trunk:


https://user-images.githubusercontent.com/16275880/185911406-e4449661-1199-49db-87b7-9220ba1aa261.mov